src/Database/Commands/PatchByIndexCommand.ts

```diff
@@ -0,0 +1,45 @@
+import {IndexQueryBasedCommand} from "./IndexQueryBasedCommand";
+import {IndexQuery} from "../Indexes/IndexQuery";
+import {PatchRequest} from "../../Http/Request/PatchRequest";
+import {QueryOperationOptions} from "../Operations/QueryOperationOptions";
+import {RequestMethods} from "../../Http/Request/RequestMethod";
+import {IResponse, IResponseBody} from "../../Http/Response/IResponse";
+import {IRavenResponse} from "../RavenCommandResponse";
+import {StatusCode, StatusCodes} from "../../Http/Response/StatusCode";
+import {ErrorResponseException, InvalidOperationException} from "../DatabaseExceptions";
+import {StringUtil} from "../../Utility/StringUtil";
+import {ServerNode} from "../../Http/ServerNode";
+
+export class PatchByIndexCommand extends IndexQueryBasedCommand {
+  protected patch?: PatchRequest = null;
+
+  constructor(indexName: string, queryToUpdate: IndexQuery, patch?: PatchRequest, options?: QueryOperationOptions) {
+    super(RequestMethods.Patch, indexName, queryToUpdate, options);
+    this.patch = patch;
+  }
+
+  public createRequest(serverNode: ServerNode): void {
+    if (!(this.patch instanceof PatchRequest)) {
+      throw new InvalidOperationException('Patch must me instanceof PatchRequest class');
+    }
+
+    this.payload = this.patch.toJson();
+    this.endPoint = StringUtil.format('{url}/databases/{database}', serverNode,this.params);
```

Comment:
buildPath implemented in IndexQueryBasedCommand (abstract class which is base for PatchByIndexCommand and DeleteByIndexCommand)